Admin/dataobjects/AdminUser.php

```diff
@@ -374,6 +382,41 @@ protected function loadHistory()
 	}
 
 	// }}}
+	// {{{ protected function loadMostRecentLogin()
+
+	/**
+	 * Gets most recent login history for this user
+	 *
+	 * If instance is set with the {@link AdminUser::setInstance()} method,
+	 * history will be limited that that instance.
+	 *
+	 * @return AdminUserHistory a {@link AdminUserHistory} containing
+	 *                           this admin user's most recent login history.
+	 *
+	 * @see AdminUser::setInstance()
+	 */
+	protected function loadMostRecentLogin()
```

Comment:
Rather than a magic loader, this method could be a public method that accepts a nullable SiteInstance. Then you don't need to manage instance state on the admin user.